packages/core/src/stateUtils.ts

```diff
@@ -1842,27 +1827,41 @@ export function macrostep<
   const { _internalQueue } = nextState;
   let maybeNextState = nextState;
 
-  while (_internalQueue.length && !maybeNextState.done) {
-    const _previousEvent = maybeNextState._event;
-    const raisedEvent = _internalQueue.shift()!;
-    const currentActions = maybeNextState.actions;
-
-    maybeNextState = machine.microstep(
+  while (!maybeNextState.done) {
+    const enabledTransitions = selectEventlessTransitions(
```

Comment:
`enabledTransitions` sounds like **any** enabled transitions
```suggestion
    const eventlessTransitions = selectEventlessTransitions(
```

Reply:
Agree, will change - this is just copied from the SCXML algorithm